RecoEgamma/EgammaTools/src/GBRForestTools.cc

```diff
@@ -0,0 +1,112 @@
+#include "RecoEgamma/EgammaTools/interface/GBRForestTools.h"
+
+#include <iostream>
+#include <fstream>
+
+namespace {
+
+  // From https://stackoverflow.com/questions/18972258/index-of-nth-occurrence-of-the-string
+  int strpos(std::string haystack, char needle, int nth)
```

Comment:
Ok, the "i" and "nth" variable are now unsigned ints. However, I didn't change anything else, as your implementation didn't work.

Reply:
(NB: this is not yet the full review, that will follow later)

I finally tested the implementation that I scribbled above. With a few little adjustments, it works:
```
int strpos(std::string & haystack, char needle, unsigned int nth)
  {// Will return position of n-th occurence of a char in a string.
    int found = 0;
    for (unsigned int i=0 ; i<nth ; ++i) {
        std::size_t pos = haystack.find(needle, found);
        if (pos == std::string::npos) return -1; 
        else found = pos+1;
    }
    return found;
  }
```

I tested it together with your code, and yours seems to give some odd result in the "pathological" cases (probably not needed here as the nch inputs are always hardcoded when the function is called). 

In the following example I looked for a character which was present three times in a string.  I wrote out the following:
```
  n0 = strpos(str, chr2, 0);
  n1 = strpos(str, chr2, 1);
  n2 = strpos(str, chr2, 2);
  n3 = strpos(str, chr2, 3);
  n4 = strpos(str, chr2, 4);
    
  std::cout << "- NEW/OLD:  0 -> " << n0 << " , 1 -> " << n1 << " , 2 -> " << n2 << " , 3 -> " << n3 << " , 4 -> " << n4 << std::endl;
```

and the output says:
```
- OLD:  0 -> -1 , 1 -> 15 , 2 -> 24 , 3 -> 45 , 4 -> 45
- NEW:  0 -> 0 , 1 -> 15 , 2 -> 24 , 3 -> 45 , 4 -> -1
```

Differences:
- For nch=0 your code returns 1 and mine 0 (this can be modified, but I think it makes sense as such)
- For nch larger than the maximum number of occurrencies of the character in the string your code always returns the  position of the last occurrency, which is odd

Taking into account that your implementation is more complicate and requires several copies of strings, I would move on what I suggest (or something even simpler, that can be probably invented or found).